### ISSUE TITLE
[1.12.2] Texture of Futura Blocks is messed up 

### ISSUE THREAD

**Issue by Draskh** [Chisel-Team/Chisel#618]
The texture is flickering. The gray and cyan variants are affected.

![2017-12-28_11 05 19](https://user-images.githubusercontent.com/26455012/34412415-66cf940a-ebdd-11e7-87fd-68284b764ee7.png)

**Comment by tterrag1098** [comment]
Optifine?

**Comment by Draskh** [comment]
OptiFine 1.12.2 HD U C7

**Comment by tterrag1098** [comment]
Try without

**Comment by Draskh** [comment]
Chisel - MC1.12-0.1.1.24
CTM - MC1.12-0.2.3.9 
without Optifine
![2017-12-30_01 38 47](https://user-images.githubusercontent.com/26455012/34450175-e9635f02-ed02-11e7-95cc-c6d86815c3f3.png)

**Comment by tterrag1098** [comment]
I have never seen this issue present without Optifine, and certainly can't reproduce it myself. Could you make a test world, and try removing mods? I can only assume this is some kind of bad interaction with another mod.

**Comment by Draskh** [comment]
I will check this.

**Comment by chunkedelement** [comment]
We are getting this on 1.12.2 with the grey screen texture futura block. We are using Chisel mod version 1.12.2-0.2.0.31/CTM-1.12.2-0.3.0.15, Forge-14.23.4.2705, in a custom mod-pack on curse. The blue screen ones work fine. Does the Better-FPS mod interfere with this texture? Not using optifine. Thanks 
https://minecraft.curseforge.com/projects/lighter-elements/relations/dependencies
is a link to the mod dependencies that are in the pack.

![image](https://user-images.githubusercontent.com/34798853/42256466-861ca5ca-7f1f-11e8-8c90-47544def3300.png)
![image](https://user-images.githubusercontent.com/34798853/42256472-93410ca0-7f1f-11e8-94a0-6738cc0f8a0d.png)

**Comment by katubug** [comment]
Hi, we're having the same z-fighting on our server. Here is our modlist with current versions: https://docs.google.com/spreadsheets/d/1XS4oFqjvtOMSajuKgJ6v1MvM3L1FD6YEaLbAYtJertg/edit?usp=sharing

![2019-04-28_19 22 36](https://user-images.githubusercontent.com/20935286/56872136-36da0680-69eb-11e9-8bd4-58c9f17bb4a8.png)

We don't use OptiFine, but we do have BetterFPS, as the other user mentioned. Tomorrow I'll do some testing.

**Comment by katubug** [comment]
Okay, sorry, I know I said I'd do testing a few days ago but I've finally gotten to it.

The first thing I noticed is that, even with all mods removed, the grey screen block is not animated. There's no z-fighting, but the animated texture only appears on the cyan. The grey block has a plain black screen. Not sure if that's intentional. Image of what I mean:

![2019-05-03_07 59 09](https://user-images.githubusercontent.com/20935286/57138864-58374b80-6d79-11e9-82a9-0424b0547efe.png)

After some testing, it seems that the z-fighting issue is tied to large modlists? Here is how I came to that conclusion:

- I created a new modpack with the same version of MC (1.12.2) and Forge (forge-1.12.2-14.23.5.2825-universal) and Chisel (Chisel-MC1.12.2-0.2.1.35)/CTM (CTM-MC1.12.2-0.3.3.22) that I have on my main server.
- In groups of ~5, I added in the mods from my modpack. The z-fighting did not occur until I reached Rustic.
- I pulled all the mods out except Rustic, Chisel, and CTM - the z-fighting was gone. So I added back in a handful of mods at a time, tested them, then removed them and added in the next handful. Using this method, I worked my way through every mod in the folder without encountering the z-fighting.
- At this point I was beginning to suspect that Rustic did not have anything to do with it, but rather that it was a numbers issue. I put all the mods I'd tested back in and encountered the z-fighting.
- Then I removed Rustic and threw in another handful of mods. Lo and behold, the z-fighting returned. Removing them fixed the issue again.
- I am now testing to see what the number of mods is that seems to be breaking the texture.

**Comment by katubug** [comment]
Okay! After roughly three hours of testing, I think the magic number is 188 mods (as listed by forge on the main Minecraft screen)/155 items in the mod folder. Adding even one mod to that causes the z-fighting. I have absolutely no idea why this would be the case, but it's the only conclusion I can draw from my testing.

I really hope this helps.

**Comment by katubug** [comment]
Okay, well I just loaded an old save with 167 files in the mod folder and there's no z-fighting. So I have no idea. I give up, lol.

**Comment by Draskh** [comment]
Could you give me your modlist? I will check for same mods.

**Comment by katubug** [comment]
Yes, I linked it above: https://docs.google.com/spreadsheets/d/1XS4oFqjvtOMSajuKgJ6v1MvM3L1FD6YEaLbAYtJertg/edit?usp=sharing

**Comment by chunkedelement** [comment]
Thanks for all your efforts. I have rustic in a few packs as well and this issue is still there as of the past update. Is rustic somehow the issue? The LE pack has only 126 mods so not sure if that is the z-fighting cause as its under 150. I will try to test again and post back. Here is the modlist of the LE pack on curse, https://minecraft.curseforge.com/projects/lighter-elements/relations/dependencies
Thanks.

**Comment by tterrag1098** [comment]
Chasing mod counts is a red herring. It's either one specific mod causing it, or it's not related to mods at all. A more productive line of questioning might be to say what GPU you have (check F3 screen). If everyone affected is on one type of card (Intel/AMD/Nvidia) it's probably a hardware issue.

**Comment by katubug** [comment]
Yeah, as my followup said, it definitely wasn't the mod count. It is weird
that it'd only occur sporadically if it's a hardware issue though. But if
it helps, I have a GeForce GTX 970.

**Comment by Draskh** [comment]
As you all can see, I have 940M in the screenshots, 
I will check if it happens with GTX 650 now.

**Comment by Draskh** [comment]
Seems broken on GTX 650 also. Here is a list of mods if anyone might see any similaritie.
[mods.txt](https://github.com/Chisel-Team/Chisel/files/3206390/mods.txt)

![2019-05-22_10 21 33](https://user-images.githubusercontent.com/26455012/58158775-6216e500-7c7b-11e9-9f8c-e27f68ec9cc7.png)

**Comment by Draskh** [comment]
Accidentally, I removed Tinkers' Compendium, formerly known as Tinkers' Defense. Issue seems to be fixed now.

![2019-05-22_13 59 59](https://user-images.githubusercontent.com/26455012/58172938-3c4d0880-7c9a-11e9-9b7a-2d8077851523.png)

**Comment by chunkedelement** [comment]
I have a gtx980 if that helps, so is it driver related then on how its rendered?

**Comment by tterrag1098** [comment]
I have no idea if it's hardware related, but that seems unlikely as I dev and test on a GTX770. I doubt Tinkers' Compendium is at fault, unless adding it back reliably triggers the issue again?

**Comment by katubug** [comment]
I do not have Tinker's Compendium in my modlist, so I'm not sure it's that. I'm sorry this isn't better formatted, but my current modlist is [here](https://minecraft.curseforge.com/projects/enthusiasticraft/relations/dependencies).

**Comment by Draskh** [comment]
@katubug You can place [this](https://pastebin.com/J54zX7t9) if saved as .bat in your mod folder and run then. It will print out a .txt file with contents of the folder in which it has been placed. You will need to edit a bit, but it will be formatted well.

**Comment by Draskh** [comment]
@tterrag1098 I made a test. I placed all blocks without [Tinkers' Compendium](https://minecraft.curseforge.com/projects/tinkers-compendium) again. I had no bugs. Then I added it back. The bug returned.

![2019-05-24_03 15 06](https://user-images.githubusercontent.com/26455012/58297016-6277c300-7dd6-11e9-8839-82f4b76e3c52.png)
![2019-05-24_03 42 29](https://user-images.githubusercontent.com/26455012/58297020-64da1d00-7dd6-11e9-9678-e6b52c7597b3.png)

**Comment by Draskh** [comment]
After I joined the test world again for some work with CraftTweaker, the futura blocks were broken although I did not change any mod. This means the bug is still sporadically.

**Comment by katubug** [comment]
Here is a link to my full modlist:

[mods.txt](https://github.com/Chisel-Team/Chisel/files/3219056/mods.txt)

**Comment by ghost** [comment]
![2019-07-23_21 56 24](https://user-images.githubusercontent.com/22541077/61718093-d7468980-ad94-11e9-8a60-e16a42193a32.png)
z fighting problems in enigmatica 2 1.68a
modlist:
[modlist.txt](https://github.com/Chisel-Team/Chisel/files/3422446/modlist.txt)
i tried removing betterfps,foamfix and disable forge's threaded chunk loading but problem still persists
i am also using intel's integrated graphics(hd 4600)

**Comment by shinji257** [comment]
Old bug report but while this issue didn't show up with StoneBlock 2 I'm noticing it with FTB Revelations.  The key part is that Tinker's Compendium (or Tinker's Defense) is not present in this specific pack.

**Comment by OmegaVolta** [comment]
Through a lengthy process of trial and error, I discovered that less lag was causing the problem. I removed that, and it worked.

**Comment by Vulant** [comment]
Also having the same issue all of a sudden, I have no idea what's causing it.

**Comment by Draskh** [comment]
This issue appears and reappears again and again, without changing any mods. Also, this bug tends to disappear. Thus, I do doubt that it is related to another mod.
It is more likely a strange interaction between vanilla minecraft, forge or graphic drivers.

**Comment by ghost** [comment]
Do we have now a fix for that ?
